lib/pushmi_pullyu/aip/solr_fetcher.rb

```diff
@@ -1,47 +1,33 @@
-require 'pushmi_pullyu'
-require 'net/http'
 require 'json'
+require 'net/http'
 
 class PushmiPullyu::AIP::SolrFetcher
 
+  class SolrFetchError < StandardError; end
+
   def initialize(noid)
     @noid = noid
   end
 
   def fetch_permission_object_ids
-    json = run_query_json("accessTo_ssim:#{@noid}", fields: 'id', url_extra: nil)
-    hash = JSON.parse(json)
+    hash = JSON.parse(run_query_json)
```

Comment:
Heavily simpfy this. Don't need to string interpolate most of this. And most of this stuff will never change like fileds `id`, as this is the only place this method is being called. Use instance variable noid instead of passing it in as an arugment.